From c990345eb83cba8e1606f643b19b99520e54e06c Mon Sep 17 00:00:00 2001
From: Eli Schwartz <eschwartz@gentoo.org>
Date: Mon, 11 Nov 2024 16:15:24 -0500
Subject: update set for "rust-rebuild" to adapt to changes in gentoo

The virtual package is deprecated and we use an eclass to depend on an
any-of group for the relevant rust compilers. This set no longer does
anything without an update.

Start checking BDEPEND for the specific rust compilers. Also keep
checking for the virtual, so that "rust-rebuild" reports on packages
that have not, in fact, been rebuilt yet.

Bug: https://bugs.gentoo.org/906044
Closes: https://github.com/gentoo/portage/pull/1399
Signed-off-by: Eli Schwartz <eschwartz@gentoo.org>
Signed-off-by: Mike Gilbert <floppym@gentoo.org>
---
 cnf/sets/portage.conf | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cnf/sets/portage.conf b/cnf/sets/portage.conf
index c272f98db..fd78364c7 100644
--- a/cnf/sets/portage.conf
+++ b/cnf/sets/portage.conf
@@ -110,8 +110,8 @@ class = portage.sets.dbapi.UnavailableBinaries
 [changed-deps]
 class = portage.sets.dbapi.ChangedDepsSet
 
-# Installed packages for which vdb *DEPEND includes virtual/rust
+# Installed packages for which vdb *DEPEND includes rust
 [rust-rebuild]
 class = portage.sets.dbapi.VariableSet
 variable = BDEPEND
-includes = virtual/rust
+includes = dev-lang/rust dev-lang/rust-bin virtual/rust
-- 
cgit v1.2.3-65-gdbad

